tendrl/commons/utils/log_utils.py

```diff
@@ -0,0 +1,41 @@
+"""Handles logging functionality."""
+
+import sys
+from tendrl.commons.event import Event
+from tendrl.commons.message import Message
+from inspect import getframeinfo
+from inspect import stack
+
+def log(log_priority, publisher_id, log_payload, job_id=None,
+        flow_id=None, parent_id=None, cluster_id=None,
+        message_id=None, timestamp=None, node_id=None,
+        caller=None, *args, **kwargs):
+    """
+    Function used for logging errors/output/info.
+    Args:
+        log_priority [Type : String]: Priority of the Log Message (error/info)
+        publisher_id [Type : Integer] : Id of publisher (mandatory)
+        log_payload [Type: Dict] : Payload can contain /
+                                   parameters like message that is to be logged
+    """
+    if caller is None:
```

Comment:
And I guess I should check if caller is None, and pass the caller to the Message function as well?